File: app/src/main/java/com/ayat/candyapp/di/UiControllerComponent.java
package com.ayat.candyapp.di;


import com.blessedtreeit.product.activitycenter.flows.add_task.AddTaskActivity;
import com.blessedtreeit.product.activitycenter.flows.add_task.AssignedToSelectionFragment;
import com.blessedtreeit.product.activitycenter.flows.entry.EntryActivity;
import com.blessedtreeit.product.activitycenter.flows.login.LoginActivity;
import com.blessedtreeit.product.activitycenter.flows.nav_drawer.NavActivity;
import com.blessedtreeit.product.activitycenter.flows.tasks.TasksFragment;
import dagger.Component;

@ViewScope
@Component(
        dependencies = {ApplicationComponent.class},
        modules = {UiControllerModule.class, FactoryModule.class}
)
public interface UiControllerComponent {

    void inject(LoginActivity loginActivity);

    void inject(EntryActivity entryActivity);

    //  void inject(BaseFragment baseFragment);

    void inject(NavActivity navActivity);

    void inject(TasksFragment tasksFragment);

    void inject(AddTaskActivity addTaskActivity);

    void inject(AssignedToSelectionFragment assignedToSelectionFragment);
}

File: app/src/main/java/com/ayat/candyapp/di/UiControllerComponent.kt
package com.ayat.candyapp.di


import com.ayat.candyapp.user_flow.home.HomeActivity
import com.ayat.candyapp.user_flow.login.LoginActivity
import com.ayat.candyapp.user_flow.signup.SignupActivity
import dagger.Component
/**
 *Created by Ayat Khriasat on 27,April,2019 at 8:32 PM
 *Email: ayatzkhraisat@gmail.com
 *Project: CandyApp
 **/
@ViewScope
@Component(dependencies = [ApplicationComponent::class], modules = [UiControllerModule::class,ViewModelsModule::class])
interface UiControllerComponent {

    fun inject(loginActivity: LoginActivity)
    fun inject(homeActivity: HomeActivity)
    fun inject(signupActivity: SignupActivity)

}